flow_pre_lda.scala

```diff
@@ -134,18 +142,139 @@ def bin_column(value: String, cuts: Array[Double]) = {
+
+    val srcIP : String = row(srcIP_feedback_index)
+    val dstIP : String  = row(dstIP_feedback_index)
+    val sport : String = row(sport_feedback_index)
+    val dport : String = row(dport_feedback_index)
+    val tstart : String = row(tstart_feedback_index)
+    val ipkts : String = row(ipkt_feedback_index)
+    val ibyts : String = row(ibyt_feedback_index)
+
+
+    // it is assumed that the format of the time object coming from the feedback is
+    //                  YYYY-MM-DD HH:MM:SS
+    //   for example:   2016-04-21 03:58:13
+    val hourMinSecond : Array[String] = tstart.split(' ')(1).split(':')  // todo: error handling if the line is malformed
+    val hour = hourMinSecond(0)
+    val min  = hourMinSecond(1)
+    val sec = hourMinSecond(2)
+
+
+    val time_flow_index = 0
+    val year_flow_index = 1
+    val month_flow_index = 2
+    val day_flow_index = 3
+    val hour_flow_index = 4
+    val minute_flow_index = 5
+    val second_flow_index = 6
+    val tdur_flow_index = 7
+    val sip_flow_index = 8
+    val dip_flow_index = 9
+    val sport_flow_index = 10
+    val dport_flow_index = 11
+    val proto_flow_index = 12
+    val flag_flow_index = 13
+    val fwd_flow_index = 14
+    val stos_flow_index = 15
+    val ipkt_flow_index  = 16
+    val ibyt_flow_index = 17
+    val opkt_flow_index = 18
+    val obyt_flow_index = 19
+    val input_flow_index= 20
+    val output_flow_index = 21
+    val sas_flow_index = 22
+    val das_flow_index = 23
+    val dtos_flow_index = 24
+    val dir_flow_index  = 25
+    val rip_flow_index = 26
+
+
+    val buf  = new StringBuilder
+    for (i <- 0 to 26) {
+        if ( i == hour_flow_index) {
+            buf ++= hour
+        } else if (i == minute_flow_index) {
+            buf ++= min
+        } else if (i == second_flow_index) {
+            buf ++= sec
+        } else if (i == ipkt_flow_index) {
+            buf ++= ipkts
+        } else if (i == ibyt_flow_index) {
+            buf ++= ibyts
+        } else if (i == sport_flow_index) {
+            buf ++= sport
+        } else if (i == dport_flow_index) {
+            buf ++= dport
+        } else if (i == sip_flow_index) {
+            buf ++= srcIP
+        } else if ( i == dip_flow_index) {
+            buf ++= dstIP
+        } else {
+            buf ++= "##"
+        }
+        if (i < 26) {
+            buf + ','
+        }
+    }
+    buf.toString()
+}
+val rawdata : RDD[String] = sc.textFile(file)
+val datanoheader : RDD[String] = removeHeader(rawdata)
+
+
+val scoredFileExists = new java.io.File(scoredFile).exists
 
-val rawdata = sc.textFile(file)
+val scoredData : Array[String] = if (scoredFileExists) {
+    val duplicationFactor = System.getenv("DUPFACTOR").toInt
 
-val datanoheader = removeHeader(rawdata)
-val datagood = datanoheader.filter(line => line.split(",").length == 27)
+    val nonThreateningRows = Source.fromFile(scoredFile).getLines().toArray.drop(1).filter(l=>(l.split(',')(0).toInt == 3))
```

Comment:
can we pick a more neutral name here?  the code is intended to be generic and named in terms of what it does, not the domain itself